### PR TITLE
feat(`consensus`): impl `SignableTx` for `TxDeposit`

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Ref: https://github.com/foundry-rs/foundry/pull/9047 

## Solution

impl `SignableTransaction<alloy_primitives::Signature>` for `TxDeposit`

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/consensus/src/transaction/deposit.rs [hunk 1]
```diff
@@ -241,6 +243,28 @@ impl Decodable for TxDeposit {
     }
 }
 
+impl SignableTransaction<alloy_primitives::Signature> for TxDeposit {
```
Comment (emhane) [inline crates/consensus/src/transaction/deposit.rs:246]:
```suggestion
impl SignableTransaction<Signature> for TxDeposit {
```
nitpick, bring this into scope in module import prelude

File: crates/consensus/src/transaction/deposit.rs [hunk 2]
```diff
@@ -241,6 +243,28 @@ impl Decodable for TxDeposit {
...
+
+    fn payload_len_for_signature(&self) -> usize {
+        self.encoded_len(false)
+    }
+
+    fn encode_for_signing(&self, out: &mut dyn alloy_rlp::BufMut) {
+        self.encode_inner(out, false);
+    }
+
+    fn into_signed(self, signature: alloy_primitives::Signature) -> alloy_consensus::Signed<Self> {
```
Comment (emhane) [inline crates/consensus/src/transaction/deposit.rs:257]:
```suggestion
    fn into_signed(self, signature: Signature) -> alloy_consensus::Signed<Self> {
```
nitpick

**Review by emhane — APPROVED** [review]

File: crates/consensus/src/transaction/deposit.rs [hunk 3]
```diff
@@ -241,6 +243,28 @@ impl Decodable for TxDeposit {
...
+    fn encode_for_signing(&self, out: &mut dyn alloy_rlp::BufMut) {
+        self.encode_inner(out, false);
+    }
+
+    fn into_signed(self, signature: Signature) -> alloy_consensus::Signed<Self> {
+        let payload_length = self.encoded_len(false) + signature.rlp_vrs_len();
+        let mut out = Vec::with_capacity(payload_length);
+        self.encode_inner(&mut out, false);
+        signature.write_rlp_vrs(&mut out);
+        let hash = keccak256(out);
```
Comment (klkvr) [inline crates/consensus/src/transaction/deposit.rs:262]: this will produce invalid hash. deposit transaction rlp encoding does not include a signature

**Review by klkvr — COMMENTED** [review]
deposit transactions are not "signable" and do not carry a signature, why do we need this?

**Comment by klkvr** [comment]
I see, we migrated this from anvil

I don't think we need this for it because signature is always getting dropped when converting to envelope https://github.com/foundry-rs/foundry/blob/47f1ecb9c6f7e251c5bf2452c1f327d5508481a9/crates/anvil/src/eth/sign.rs#L143

imo even if it's needed in anvil we shouldn't introduce trait implementations which produce off-spec encodings

**Comment by yash-atreya** [comment]
Yeah @klkvr, my mistake. 

Revert PR: https://github.com/alloy-rs/op-alloy/pull/153

**Pull request merged**